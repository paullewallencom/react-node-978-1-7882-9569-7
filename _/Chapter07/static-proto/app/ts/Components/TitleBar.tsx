import * as React from "react";
import { remote } from "electron";
import { Header, Navigation, Icon } from "react-mdl";

export default class TitleBar extends React.Component<{}, {}> {

  private onClose = () => {
    remote.getCurrentWindow().close();
  }
  render() {
    return (
     <Header title="RSS Aggregator" scroll>
        <Navigation>
            <a href="#" onClick={this.onClose}><Icon name="close" /></a>
        </Navigation>
    </Header>
    );
  }
}
