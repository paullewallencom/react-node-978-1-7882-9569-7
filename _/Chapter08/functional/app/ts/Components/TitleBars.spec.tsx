import * as React from "react";
import TitleBar from "./TitleBar";
import * as renderer from "react-test-renderer";

describe( "TitleBar", () => {
  it( "renders correctly", () => {
    const tree = renderer.create(
      <TitleBar />
    ).toJSON();
    expect( tree ).toMatchSnapshot();
  });
});

