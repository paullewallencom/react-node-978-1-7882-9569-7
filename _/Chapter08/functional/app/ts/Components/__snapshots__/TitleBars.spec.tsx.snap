// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar renders correctly 1`] = `
<header
  className="mdl-layout__header mdl-layout__header--scroll"
>
  <div
    className="mdl-layout__header-row"
  >
    <span
      className="mdl-layout-title"
    >
      RSS Aggregator
    </span>
    <div
      className="mdl-layout-spacer"
    />
    <nav
      className="mdl-navigation"
    >
      <a
        className="mdl-navigation__link"
        href="#"
        onClick={[Function]}
      >
        <i
          className="material-icons"
        >
          close
        </i>
      </a>
    </nav>
  </div>
</header>
`;
